include/Reader/readerir.h

```diff
@@ -1515,6 +1540,7 @@ class GenIR : public ReaderBase {
   ///                                  and the array itself.
   uint32_t addArrayFields(
       std::vector<llvm::Type *> &Fields, bool IsVector, uint32_t ArrayRank,
+      llvm::Type *ElementTy,
```

Comment:
I think this signature is confusing. I would prefer to have just ElementTy parameter and remove ElementCorType and ElementClassHandle and have the caller always calculate ElementTy. That should also allow you to remove GetElementAggregateFields and DeferredDetailClasses.